src/tribler-core/tribler_core/modules/tests/test_versioncheck.py

```diff
@@ -97,3 +97,40 @@ async def test_non_json_response(self):
             versioncheck_manager.check_failed = True
 
         self.assertTrue(versioncheck_manager.check_failed)
+
+    @timeout(10)
+    async def test_version_check_timeout(self):
+        await self.setup_version_server(json.dumps({'name': 'v1337.0'}))
+
+        self.new_version_called = False
+        # Default timeout is 5 seconds so under normal circumstance, we don't expect timeout
+        await self.session.version_check_manager.check_new_version()
+        self.assertTrue(self.new_version_called)
+
+        # Setting a timeout of 1ms, version checks should fail
+        versioncheck_manager.VERSION_CHECK_TIMEOUT = 0.001
+
+        self.new_version_called = False
+        await self.session.version_check_manager.check_new_version()
+        self.assertFalse(self.new_version_called)
+
+        await self.site.stop()
```

Comment:
Cleanup stuff must be in the `tearDown` method. Otherwise, the line will not be called on test failure and the failed test can affect the rest of the tests that will run in the same process

Reply:
Good point